libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -298,9 +298,38 @@ void CompilerUtils::zeroInitialiseMemoryArray(ArrayType const& _type)
 	m_context << Instruction::SWAP1 << Instruction::POP;
 }
 
-void CompilerUtils::memoryCopy()
+void CompilerUtils::memoryCopy(bool _useIdentityPrecompile)
 {
 	// Stack here: size target source
+
+	if (!_useIdentityPrecompile)
+	{
+		m_context.appendInlineAssembly(R"(
+			{
+			// expects three locals: src, dst, len
+
+			// copy 32 bytes at once
+			start32:
+			jumpi(end32, lt(len, 32))
+			mstore(dst, mload(src))
+			dst := add(dst, 32)
+			src := add(src, 32)
+			len := sub(len, 32)
+			jump(start32)
+			end32:
+
+			// copy the last byte
```

Comment:
The comment is wrong, it copies the remainder (0 < n < 32 bytes). Copying a single byte indeed would be simple.